air/src/stack/field_ops/mod.rs

```diff
@@ -0,0 +1,190 @@
+    // constraints.
+
+    // There's one unique constraint in INCR field operation in which 1 is added to the top
+    // element of the stack.
+    8,
+    // There's one unique constraint in INV field operation in which the top element in the
+    // stack is replaced with it's inverse.
+    9,
+    // There's one unique constraint in NEG field operation in which the top element in the
+    // stack is replaced with it's negation.
+    8,
+    // There are two unique operations in NOT field operation. The first one checks if the
+    // top element in the stack is a boolean or not whereas the second verifies if the second
+    // first element in the next trace is a boolean not of first element in the current trace.
+    9, 8,
+];
+
+// FIELD OPERATIONS TRANSITION CONSTRAINTS
+// ================================================================================================
+
+/// Builds the transition constraint degrees for the field operations.
+pub fn get_transition_constraint_degrees() -> Vec<TransitionConstraintDegree> {
+    CONSTRAINT_DEGREES
+        .iter()
+        .map(|&degree| TransitionConstraintDegree::new(degree))
+        .collect()
+}
+
+/// Returns the number of transition constraints for the Field operations.
+pub fn get_transition_constraint_count() -> usize {
+    NUM_CONSTRAINTS
+}
+
+/// Enforces constraints for the Field operations.
+pub fn enforce_constraints<E: FieldElement>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+    op_flag: &OpFlags<E>,
+) -> usize {
+    let mut index = 0;
+
+    // fetching the respective flags of field operations from op_flag struct.
+    let incr_op_flag = op_flag.incr();
+    let inv_op_flag = op_flag.inv();
+    let neg_op_flag = op_flag.neg();
+    let not_op_flag = op_flag.not();
+
+    // Enforce unique constaints of the INCR operation.
+    index += enforce_incr_unique_constraints(frame, result, incr_op_flag);
+    // Enforce unique constaints of the INV operation.
+    index += enforce_inv_unique_constraints(frame, &mut result[index..], inv_op_flag);
+    // Enforce unique constaints of the NEG operation.
+    index += enforce_neg_unique_constraints(frame, &mut result[index..], neg_op_flag);
+    // Enforce unique constaints of the NOT operation.
+    index += enforce_not_unique_constraints(frame, &mut result[index..], not_op_flag);
+
+    index
+}
+
+// TRANSITION CONSTRAINT HELPERS
+// ================================================================================================
+
+/// Enforces unique constraints of the INCR operation. The INCR operation increments the
+/// top element in the stack by 1. Therefore, the following constraints are enforced:
+/// - The next element in the next frame should be equal to the addition of frist element in the
+///   current frame and 1.
+pub fn enforce_incr_unique_constraints<E: FieldElement>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+    op_flag: E,
+) -> usize {
+    let mut constraint_offset = 0;
```

Comment:
nit: given that these constraints are pretty simple, I'm not sure we need to track constraint offset explicitly. We could just return `1` at the end of the function.

This applies to other functions as well.

Reply:
Updated the implementation